test/integration/acceptance/tx_heavy_data.cpp

```diff
@@ -0,0 +1,209 @@
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      // "foo" transactions will be not passed because it has large size into
+      // one field - 5Mb per one set
+      .sendTx(complete(setAcountDetailTx("foo", generateData(5 * 1024 * 1024))))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * NOTE: test is disabled until fix of
+ * https://soramitsu.atlassian.net/browse/IR-1205 will be not completed.
+ * @given some user with all required permissions
+ * @when send many txes with addAccountDetail with large data inside
+ * @then transaction have been passed
+ */
+TEST_F(HeavyTransactionTest, DISABLED_ManyLargeTxes) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .sendTx(
+          complete(setAcountDetailTx("foo_1", generateData(2 * 1024 * 1024))))
+      .sendTx(
+          complete(setAcountDetailTx("foo_2", generateData(2 * 1024 * 1024))))
+      .sendTx(
+          complete(setAcountDetailTx("foo_3", generateData(2 * 1024 * 1024))))
+      .sendTx(
+          complete(setAcountDetailTx("foo_4", generateData(2 * 1024 * 1024))))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 4); })
+      .done();
+}
+
+/**
+ * NOTE: test is disabled until fix of
+ * https://soramitsu.atlassian.net/browse/IR-1205 will be not completed.
+ * @given some user with all required permissions
+ * @when send tx with many addAccountDetails with large data inside
+ * @then transaction is passed
+ */
+TEST_F(HeavyTransactionTest, DISABLED_VeryLargeTxWithManyCommands) {
+  auto big_data = generateData(3 * 1024 * 1024);
+  auto large_tx_builder = setAcountDetailTx("foo_1", big_data)
+                              .setAccountDetail(kUserId, "foo_2", big_data)
+                              .setAccountDetail(kUserId, "foo_3", big_data);
+
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      // in itf tx build from large_tx_build will be pass Torii but in
+      // production the transaction will be failed before stateless validation
+      // because of size.
+      .sendTx(complete(large_tx_builder))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * NOTE: test is disabled until fix of
+ * https://soramitsu.atlassian.net/browse/IR-1205 will be not completed.
+ * @given some user with all required permissions
+ * AND itf that hanlde only 1 tx in OS.
+ * @when send txes with addAccountDetail with large data inside.
+ * AND transactions are passed steful validation
+ * @then transactions are passed
+ * AND query for getting is done
+ */
+TEST_F(HeavyTransactionTest, DISABLED_QueryLargeData) {
+  auto query_checker = [](auto &status) {
+    boost::apply_visitor(
```

Comment:
Also maybe check it against the `data` (so nothing was truncated)?

Reply:
Yes, good point.